R/impact_central.R

```diff
@@ -142,9 +142,17 @@ impact_by_year_of_vaccination <- function(meta1, raw_impact, fvps, fvps_updates
     stop("method2a can not accommodate both routine and campaign impact in the same time, as routine and campaign impact are calculated differently.")
   }
   # prepare data
-  fvps$time <- fvps$year - fvps$age
-  fvps <- merge_by_common_cols(fvps, vaccine_delivery, all.y = TRUE)
   fvps <- fvps[fvps$year %in% vaccination_years, ]
+  fvps$time <- fvps$year - fvps$age
+  fvps_tmp <- merge_by_common_cols(fvps, vaccine_delivery, all.y = TRUE)
+  if(all(is.na(fvps_tmp$fvps))){
```

Comment:
@xiangli313 do you have an example which hits this code we can add a test for?

Reply:
I have a feeling that this code will be working for countries with no vaccination introduction. I'll check this.